grid-compose/internal/types/types.go

```diff
@@ -0,0 +1,65 @@
+package types
+
+// Service represents a service in the deployment
+type Service struct {
+	Flist       string       `yaml:"flist"`
+	Entrypoint  string       `yaml:"entrypoint,omitempty"`
+	Environment []string     `yaml:"environment"`
+	Resources   Resources    `yaml:"resources"`
```

Comment:
you can define a new type for this which implemnt the `YamlUnmarshlar` to directly have the environment as `map[string]string` which what zmachine.envirnment expect